bookdown/01-mlr3-basics.Rmd

```diff
@@ -24,29 +24,27 @@ The `r mlr_pkg("mlr3")` package provides R6 classes for the these essential buil
 
 You will later learn how to enhance learners to full pipelines and tune hyperparameters, but for now we will stick to this simple workflow.
 
-## Quick R6 Intro for Starters
-
-R6 is one of R's more recent dialects for object-oriented programming.
-While R6 might take a little while to learn for people who have never seen OO before, you can find here what you need to know in order to use R6 with `r mlr_pkg("mlr3")` from the outside.
-
-* Objects are created from a so called "factory method" with a constructor.
-  For example `foo = Foo$new(bar = 1)` creates a new object of class `Foo`, by passing `1` to its constructor.
-* Classes encode their state via so called "member variables".
-  Assuming, the constructor of `Foo` would directly store the state of `bar = 1` in the created object in a public member variable, we could access it now via `foo$bar` and retrieve its value or change it.
-* Classes usually expose some methods to
-  * retrieve information,
-  * perform some action,
-  * change the state of the object.
-
-  For example, `foo$inc()` might increment the value of `bar` in `foo` by `+1`, and hence changes the state of the object.
-* Classes have public and private parts of their interface, regarding methods and member variables.
-  We as outside users can only access the public parts and do not have to worry about the private ones.
-* R6 classes are implemented via environments / references.
-  This means that R6 variables do not actually hold the object itself but rather a reference (pointer) to them.
-  This means that `foo2 = foo` does not copy the object, but only copies the reference, which now points to the object.
-* If you really want to fully copy an object call `foo$clone()`, possibly `foo$clone(deep = TRUE)`.
-
-If you want more details, have a look at the [R6 vignettes](https://r6.r-lib.org/).
+## Quick R6 Intro for Beginners
+
+R6 is one of R's more recent dialects for object-oriented programming (OO).
+It addresses shortcomings of earlier OO implementations in R, such as S3, which we used in `r mlr_pkg("mlr")`.
+If you are not familiar with OO programming, it might take a bit to find your way around, but if you have done OO in another language, R6 should feel familiar.
+We focus on the parts of R6 that you need to know to use `r mlr_pkg("mlr3")` here without going into further detail.
+
+* Objects are created by calling the constructor of an `R6Class()` object.
```

Comment:
```suggestion
* Objects are created by calling the constructor of a `R6Class()` object.
```

Reply:
As Bernd said on slack, a/an depends on the phonetics, not the spelling.